src/kubernetes_cluster/proof/message.rs

```diff
@@ -265,16 +333,249 @@ pub open spec fn pending_req_has_lower_req_id_than_allocator() -> StatePred<Self
     }
 }
 
-pub proof fn lemma_always_pending_req_has_lower_req_id_than_allocator()
+pub proof fn lemma_always_pending_req_has_lower_req_id_than_allocator(spec: TempPred<Self>)
+    requires
+        spec.entails(lift_state(Self::init())),
+        spec.entails(always(lift_action(Self::next()))),
     ensures
-        Self::sm_spec().entails(always(lift_state(Self::pending_req_has_lower_req_id_than_allocator()))),
+        spec.entails(always(lift_state(Self::pending_req_has_lower_req_id_than_allocator()))),
 {
     let invariant = Self::pending_req_has_lower_req_id_than_allocator();
     init_invariant::<Self>(
-        Self::sm_spec(), Self::init(), Self::next(), invariant
+        spec, Self::init(), Self::next(), invariant
     );
 }
 
+pub open spec fn ok_get_response_msg() -> FnSpec(MsgType<E>) -> bool {
+    |msg: MsgType<E>|
+        msg.src.is_KubernetesAPI()
+        && msg.content.is_get_response()
+        && msg.content.get_get_response().res.is_Ok()
+}
+
+pub open spec fn ok_get_response_msg_with_key(key: ObjectRef) -> FnSpec(MsgType<E>) -> bool {
+    |msg: MsgType<E>|
+        msg.src.is_KubernetesAPI()
+        && msg.content.is_get_response()
+        && msg.content.get_get_response().res.is_Ok()
+        && msg.content.get_get_response().res.get_Ok_0().object_ref() == key
```

Comment:
I think we can later rename `object_ref()` to `key()` or sth, to avoid confusion with another reference (owner reference)

Reply:
It's a datatype defined in kube-rs library, but we don't have to follow the name.

Reply:
I don't think this will cause confusion. Owner reference means reference of the owner, and I never omit "owner" when referring to it.